### DISCUSSION TITLE
Could not load type 'System.Data.Dataset' form assembly 'System.Data.Common'

### DISCUSSION THREAD

**Discussion by iamtorchstack** [pythonnet/pythonnet#1843] [Q&A] [answered]
### Environment

-   Pythonnet version: 3.0.0a2 PRE-RELEASE
-   Python version: 3.8.8
-   Operating System: Windows 10
-   .NET Runtime: .Net core 3.1 and .Net 5.0

### Details

-   Created a program to
> Use the DataSets from Systems.Data.Dataset

    _TODO_

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

```
  import sys, clr
  assembly_path = r".C:\User\...\...\Debug\netcoreapp3.1"
  sys.path.path.append(assembly_path)
  clr.AddReference("DatabaseControl")
  from DatabaseControl import  readConvey
 obj =  readConvey() <<<. Code failing here
```

-   System.TypeLoadException: Could not load type 'System.Data.Dataset' form assembly 'System.Data.Common', Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a

**Comment by filmor** [comment]
It does not work like this. Please read the information in the pinned ticket, I'll update the documentation in the coming weeks.

**Comment by iamtorchstack** [comment]
Hi Filmor, I do not see where I might done this wrongly. If you can, could you possibly tell me what pinned ticket you referring to?

**Comment by filmor** [comment]
There is a single pinned ticket: https://github.com/pythonnet/pythonnet/issues/984#issuecomment-778786164

**Comment by iamtorchstack** [comment]
Hi Filmor, I implemented this code as given in the ticket. First by creating the app.runtime.json config setting and calling the right DLL but I still had the error of loading 

System.TypeLoadException: Could not load type 'System.Data.Dataset' form assembly 'System.Data.Common', Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a

System.TypeLoadException: Could not load type 'System.Data.SqlClient.SqlCommand' form assembly 'System.Data.SqlClient', Version=4.6.1.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a

Please I need a possible solution or direction as I have no idea what must be done. :)

**Comment by iamtorchstack** [comment, marked as answer]
So the challenge was resolved when I reduce the .net framework generating the DLL file to 4.8. It works fine. now.  I tried with .netcore 3.1, 5 and 6 and they all did not work.
